packages/utils/src/eslint-utils/RuleCreator.ts

```diff
@@ -73,6 +74,13 @@ export function RuleCreator(urlCreator: (ruleName: string) => string) {
         },
       },
       ...rule,
+      create(context, optionsWithDefault): RuleListener {
+        if (meta.docs.requiresTypeChecking) {
```

Comment:
This is essentially an internal-only property that we use for documentation generation purposes - so we can't really rely on it like this.

Additionally this breaks usecases where we want conditional type-awareness - eg `naming-convention` only uses types if and only if you set certain options. So *always* running the util would mean the rule *always* requires parser services - even if you don't use the type-aware options.

There are also going to be usecases externally where people go against our current recommendations and do progressive enhancement of rules wherein this logic would break their usecase entirely.